Lib/fontmake/font_project.py

```diff
@@ -430,22 +431,36 @@ def save_otfs(self,
                 os.remove(otf_path)
 
     def subset_otf_from_ufo(self, otf_path, ufo):
-        """Subset a font using export flags set by glyphsLib."""
+        """Subset a font using export flags set by glyphsLib.
 
-        keep_glyphs = set(ufo.lib.get(GLYPHS_PREFIX + 'Keep Glyphs', []))
+        There are two more settings that can change export behavior:
+        "Export Glyphs" and "Remove Glyphs", which are currently not supported
+        for complexity reasons. See
+        https://github.com/googlei18n/glyphsLib/issues/295.
+        """
+        ufo_order = makeOfficialGlyphOrder(ufo)
+        ot_order = TTFont(otf_path).getGlyphOrder()
+        assert ot_order[0] == ".notdef"
+        assert len(ufo_order) == len(ot_order)
```

Comment:
ouch. this fails if the UFO does not originally contain the ".notdef" glyph. Ufo2ft always insert one when missing, but that `makeOfficialGlyphOrder(ufo)` function does not add ".notdef" if it's not already in there...
only if it's present, it forces it to be the first, otherwise it ignores it.
https://github.com/googlei18n/ufo2ft/blob/6b710f7e889368746cbcf4340cbf994998cea1ae/Lib/ufo2ft/util.py#L36-L39

in the ufo2ft.outlineCompiler, that function is called on the resulting glyphSet only after the makeMissingRequiredGlyphs method has inserted a .notdef.
But here we are calling it on the input UFO.

Reply:
We can check if '.notdef' is not in ufo, then add one at the beginning of the ufo_order list..
This could work for the time being, since the only "missing required" glyph that is being added by ufo2ft is the .notdef, but it's possible that we add more later. Relying on that implementation detail is not very future-proof.
🤔

Reply:
Ouch!

Shouldn't `makeOfficialGlyphOrder` really return the official one, with all required glyphs in it? Otherwise, maybe `makeMissingRequiredGlyphs` can be refactored into `missing_required_glyphs` and `insert_missing_required_glyphs` or somesuch? Then `makeOfficialGlyphOrder` can call the former.